fabric-combat-api-v1/src/main/java/net/fabricmc/fabric/api/combat/v1/ShotProjectileEvents.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.combat.v1;
+
+import org.jetbrains.annotations.NotNull;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.PersistentProjectileEntity;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.api.combat.v1.bow.FabricBowItem;
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public final class ShotProjectileEvents {
+	private ShotProjectileEvents() {
+	}
+
+	/**
+	 * This event modifies the projectile entity. Returning a new entity will not change the entity that is spawned in.
+	 */
+	public static final Event<ModifyProjectileFromBow> BOW_MODIFY_SHOT_PROJECTILE = EventFactory.createArrayBacked(ModifyProjectileFromBow.class, callbacks -> (bowStack, arrowStack, user, pullProgress, persistentProjectileEntity) -> {
+		for (ModifyProjectileFromBow callback : callbacks) {
+			if (callback instanceof FabricBowItem) {
+				if (callback == bowStack.getItem()) {
+					callback.modifyProjectileShot(bowStack, arrowStack, user, pullProgress, persistentProjectileEntity);
+				}
```

Comment:
Just fyi on your logic of skipping fire when the callback is a fabric bow and the handled bow is not the callback's bow. If there is exactly one callback, your invoker generation logic will be skipped. See https://github.com/FabricMC/fabric/blob/4681c750b8a508e37a08c1fe94498f21fef7e42b/fabric-api-base/src/main/java/net/fabricmc/fabric/impl/base/event/ArrayBackedEvent.java#L46-L47
Imo player shouldn't assume he can just delegate to single callback than bypass invoker logic

Reply:
i think that only happens when the invoker is registered and then future callbacks are then in future ones... i could be wrong and need to test this

Reply:
You can just write an event where the invoker generator is like
```java
callbacks -> {
  LOGGER.info("Called invoker factory with callback with length {}", callbacks.length);
  return (params) -> {
    for (CallbackName each : callbacks) {
       each.call(params);
    }
  }
}
```
and see the log outputs.

Reply:
Alright, i just realized that im already checking in the fabric bow class, i don't need to duplicate the check logic